pkg/registry/core/service/storage/rest.go

```diff
@@ -872,18 +872,20 @@ func (rs *REST) tryDefaultValidateServiceClusterIPFields(service *api.Service) e
 		}
 	}
 
-	// default families according to cluster IPs
+	// Infer IPFamilies[] from ClusterIPs[].
 	for i, ip := range service.Spec.ClusterIPs {
 		if ip == api.ClusterIPNone {
 			break
 		}
 
 		// we have previously validated for ip correctness and if family exist it will match ip family
 		// so the following is safe to do
+		//FIXME: once the 2nd-layer is removed, this could be a lie.
```

Comment:
I am not sure i understand what is 2nd layer here?

Reply:
I think he refers to this https://github.com/kubernetes/kubernetes/pull/96684

Reply:
Sorry, forgot to remove this